File: app/src/main/java/afkt/project/ui/adapter/RecyclerLoadingAdapter.java
package afkt.project.ui.adapter;

import android.graphics.drawable.Drawable;
import android.view.View;
import android.widget.FrameLayout;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import java.util.List;

import afkt.project.R;
import afkt.project.util.ViewAssistUtils;
import dev.base.DevSource;
import dev.base.widget.BaseImageView;
import dev.engine.image.DevImageEngine;
import dev.engine.image.listener.DrawableListener;
import dev.widget.assist.ViewAssist;

/**
 * detail: ViewAssist RecyclerView 适配器
 * @author Ttt
 */
public class RecyclerLoadingAdapter
        extends BaseQuickAdapter<String, BaseViewHolder> {

    public RecyclerLoadingAdapter(@Nullable List<String> data) {
        super(R.layout.adapter_recycler_loading, data);
    }

    @Override
    protected void convert(
            BaseViewHolder helper,
            String url
    ) {
        BaseImageView vid_arl_igview = helper.getView(R.id.vid_arl_igview);
        FrameLayout   vid_arl_frame  = helper.getView(R.id.vid_arl_frame);
        ViewAssist    viewAssist     = ViewAssist.wrap(vid_arl_frame);
        ViewAssistUtils.registerRecyclerLoading(viewAssist, new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loadImage(vid_arl_igview, viewAssist, url);
            }
        });
        loadImage(vid_arl_igview, viewAssist, url);
    }

    private void loadImage(
            BaseImageView imageView,
            ViewAssist viewAssist,
            String url
    ) {
        viewAssist.showIng();

        DevImageEngine.getEngine().display(
                imageView,
                url,
                new DrawableListener() {
                    @Override
                    public void onStart(DevSource source) {

                    }

                    @Override
                    public void onResponse(
                            DevSource source,
                            Drawable value
                    ) {
                        viewAssist.showSuccess();
                    }

                    @Override
                    public void onFailure(
                            DevSource source,
                            Throwable throwable
                    ) {
                        viewAssist.showFailed();
                    }
                }
        );
    }
}

File: app/src/main/java/afkt/project/ui/adapter/RecyclerLoadingAdapter.kt
package afkt.project.ui.adapter

import afkt.project.R
import afkt.project.databinding.AdapterRecyclerLoadingBinding
import afkt.project.utils.ViewAssistUtils
import android.graphics.drawable.Drawable
import android.view.ViewGroup
import dev.adapter.DevDataAdapter
import dev.base.DevSource
import dev.base.adapter.DevBaseViewBindingVH
import dev.base.adapter.newBindingViewHolder
import dev.base.widget.BaseImageView
import dev.engine.DevEngine
import dev.engine.image.listener.DrawableListener
import dev.widget.assist.ViewAssist

/**
 * detail: ViewAssist RecyclerView 适配器
 * @author Ttt
 */
class RecyclerLoadingAdapter(data: List<String>) : DevDataAdapter<String, DevBaseViewBindingVH<AdapterRecyclerLoadingBinding>>() {

    init {
        setDataList(data, false)
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): DevBaseViewBindingVH<AdapterRecyclerLoadingBinding> {
        return newBindingViewHolder(parent, R.layout.adapter_recycler_loading)
    }

    override fun onBindViewHolder(
        holder: DevBaseViewBindingVH<AdapterRecyclerLoadingBinding>,
        position: Int
    ) {
        val url = getDataItem(position)
        val viewAssist = ViewAssist.wrap(holder.binding.vidFl)
        ViewAssistUtils.registerRecyclerLoading(viewAssist) {
            loadImage(holder.binding.vidIv, viewAssist, url)
        }
        loadImage(holder.binding.vidIv, viewAssist, url)
    }

    private fun loadImage(
        imageView: BaseImageView,
        viewAssist: ViewAssist,
        url: String
    ) {
        viewAssist.showIng()
        // 加载图片
        DevEngine.getImage()?.display<Drawable>(
            imageView, url,
            object : DrawableListener() {
                override fun onStart(source: DevSource) {}
                override fun onResponse(
                    source: DevSource,
                    value: Drawable
                ) {
                    viewAssist.showSuccess()
                }

                override fun onFailure(
                    source: DevSource,
                    throwable: Throwable
                ) {
                    viewAssist.showFailed()
                }
            }
        )
    }
}